libraries/AP_Beacon/AP_Beacon_Nooploop.cpp

```diff
@@ -0,0 +1,235 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <AP_HAL/AP_HAL.h>
+#include <GCS_MAVLink/GCS.h>
+#include "AP_Beacon_Nooploop.h"
+#include <ctype.h>
+#include <stdio.h>
+
+extern const AP_HAL::HAL& hal;
+
+// constructor
+AP_Beacon_Nooploop::AP_Beacon_Nooploop(AP_Beacon &frontend, AP_SerialManager &serial_manager) :
+    AP_Beacon_Backend(frontend)
+{
+    _uart = serial_manager.find_serial(AP_SerialManager::SerialProtocol_Beacon, 0);
+    if (_uart != nullptr) {
+        _uart->begin(serial_manager.find_baudrate(AP_SerialManager::SerialProtocol_Beacon, 0));
+    }
+}
+
+// return true if sensor is basically healthy (we are receiving data)
+bool AP_Beacon_Nooploop::healthy()
+{
+    // healthy if we have parsed a message within the past 300ms
+    return ((AP_HAL::millis() - _last_update_ms) < AP_BEACON_TIMEOUT_MS);
+}
+
+// update the state of the sensor
+void AP_Beacon_Nooploop::update(void)
+{
+    // return immediately if not serial port
+    if (_uart == nullptr) {
+        return;
+    }
+
+    // check uart for any incoming messages
+    int16_t nbytes = _uart->available();
```

Comment:
or uint32_t?  This surprises me actually because we have at least 5 other drivers that use int16_t but looking at the AP_HAL_ChibiOS/UARTDriver.h's UARTDriver::available() definition it seems to be a uint32_t.

Reply:
By the way. @rmackay9 May I ask a question about AP_Beacon library here?
What is order of beacon_instance in https://github.com/ArduPilot/ardupilot/blob/f364dc933d060eb8e631dfd5ff11ab6769a88a2b/libraries/AP_Beacon/AP_Beacon_Backend.h#L44
Is it the order of pozyx beacon? (beacon_instance = 0 for origin, beacon_instance = 1 for north, beacon_instance = 2 for east)  Thank you very much
![93544158-f0929800-f98f-11ea-8ac4-983f3d1e8ac0](https://user-images.githubusercontent.com/19793511/93657440-a1b73200-fa65-11ea-9fe9-4924d7b124bd.png)